pinot-common/src/main/java/com/linkedin/pinot/common/data/DateTimeFormatSpec.java

```diff
@@ -169,22 +181,53 @@ public TimeFormat getTimeFormat() {
   }
 
   /**
-   * Extracts the simmple date format pattern from the format of a dateTimeSpec
-   * (4th token in colon separated format in case of TimeFormat=SIMPLE_DATE_FORMAT)
+   * Extracts the simple date format pattern from the format of a dateTimeSpec
+   * (4th token in colon separated format in case of TimeFormat=SIMPLE_DATE_FORMAT.
+   * If tz is also provided, need to extract only the sdf pattern part)
    * This method should not do validation of outputGranularity.
    * The validation should be handled by caller using {@link #isValidFormat(String)}
-   * eg: if format=1:HOURS:EPOCH, will throw exception
+   * eg: if format=1:HOURS:EPOCH, will return null
    * if format=1:HOURS:SIMPLE_DATE_FORMAT:yyyyMMddHH will return yyyyMMddHH
+   * if format=1:HOURS:SIMPLE_DATE_FORMAT:yyyyMMddHH tz(CST) will return yyyyMMddHH
    * @return
    */
   public String getSDFPattern() {
-    String pattern = null;
+    String sdfPattern = null;
+    // split with limit, as pattern can have ':'
+    String[] formatTokens = _format.split(COLON_SEPARATOR, MAX_FORMAT_TOKENS);
+    if (formatTokens.length == MAX_FORMAT_TOKENS) {
+      sdfPattern = formatTokens[FORMAT_PATTERN_POSITION];
+      Matcher m = SDF_PATTERN_WITH_TIMEZONE.matcher(formatTokens[FORMAT_PATTERN_POSITION]);
+      if (m.find()) {
+        sdfPattern = m.group(SDF_PATTERN_GROUP);
+      }
+    }
+    return sdfPattern;
+  }
+
+  /**
+   * Extracts the timezone from the format of a dateTimeSpec
+   * (4th token in colon separated format in case of TimeFormat=SIMPLE_DATE_FORMAT. tz is added to the end)
+   * This method should not do validation of outputGranularity.
+   * The validation should be handled by caller using {@link #isValidFormat(String)}
+   * eg: if format=1:HOURS:EPOCH, will return UTC
+   * if format=1:HOURS:SIMPLE_DATE_FORMAT:yyyyMMddHH will return UTC
+   * if format=1:HOURS:SIMPLE_DATE_FORMAT:yyyyMMddHH tz(America/Chicago) will return America/Chicago
+   * @return
+   */
+  public DateTimeZone getDateTimezone() {
```

Comment:
looks like code the duplicated. can we have another class to represent the format and encapsulate the logic in that class and have getters such as getType (SDF/EPOCH), getSDFFormat get TimeZone etc.